src/Game.js

```diff
@@ -0,0 +1,253 @@
+
+        switch (name) {
+            case 'playerOne':
+            case 'playerTwo':
+                return isValidName;
+            default:
+                break;
+        };
+    };
+
+    const isValidNumber = (name, value) => {
+        const isValid = Number(value) >= 4;
+        const isValidNInARow = Number(value) >= 2;
+
+        switch (name) {
+            case 'rows':
+            case 'columns':
+                return isValid;
+            case 'nInARows':
+                return isValidNInARow;
+            default:
+                break;
+        };
+    };
+
+    const handleName = (event) => {
+        const { name, value } = event.target;
+
+        isValidUsername(name, value);
+
+        return (
+            name === "playerOne"
+                ? setPlayerOne({ ...playerOne, name: value })
+                : setPlayerTwo({ ...playerTwo, name: value })
+        );
+    };
+
+    const handleBoardSize = (event) => {
+        const { name, value } = event.target;
+
+        isValidNumber(name, value);
+
+        switch (name) {
+            case 'rows':
+                setBoardSize({ ...boardSize, row: value });
+                break;
+            case 'columns':
+                setBoardSize({ ...boardSize, col: value });
+                break;
+            case 'nInARows':
+                setNInARow(value);
+                break;
+            default:
+                break;
+        };
+
+        return name;
+    };
+
+    const handleLoginForm = () => {
+        setViewLoginForm(true);
+    };
+
+    const handleStartGame = () => {
+        setGameStarted(true);
+        setViewBoard(true);
+    };
+
+    const renderNewRound = (newFocusRef, winner) => {
+        let ballWinner = '';
+
+        if (winner === playerOne.name) {
+            ballWinner = { isPlayerOneBall: true, isPlayerTwoBall: false };
+            setWinnerBall(ballWinner);
+        } else if (winner === playerTwo.name) {
+            ballWinner = { isPlayerOneBall: false, isPlayerTwoBall: true };
+            setWinnerBall(ballWinner);
+        };
+
+        setRestartGame(true);
+        checkPointsForPlayer(winner);
+        setWinner('');
+
+        newFocusRef && newFocusRef.current && newFocusRef.current.focus();
+    };
+
+    const checkPointsForPlayer = (winner) => {
+        let result = '';
+
+        if (winner === playerOne.name) {
+            result = playerOne.score;
+            setPlayerOne({ ...playerOne, score: playerOne.score + 1 });
+        } else if (winner === playerTwo.name) {
+            result = playerTwo.score;
+            setPlayerTwo({ ...playerTwo, score: playerTwo.score + 1 });
+        };
+
+        return result;
+    };
+
+    const runBoardCheck = (board) => {
+        const winner = getWinner(board, { playerOneName: playerOne.name, playerTwoName: playerTwo.name, nInARow: nInARow });
+        winner && setWinner(winner);
+    };
+
+    const getMoveBallComponent = (props) => {
```

Comment:
Extract these as components in separate files. Replace the magic numbers with consts